GoMeet/src/main/webapp/js/createMeetingShared.js

```diff
@@ -0,0 +1,48 @@
+const FILL_ALL_FIELDS_ALERT = 'Please fill in all required fields.'; 
+/**
+ * Takes all values from <input> tags on the page, and saves data to sessionStorage
+ * with 'name' of the <input> tag as key, and 'value' of the input tag as the value.
+ * Note:
+ *  - All <input> 'name's must be unique, otherwise the value will be overwritten.
+ *  - All <input> on the page are expected to be compulsory fields. Empty <input> will 
+ *  raise an alert requiring user to fill in every field.
+ * @returns true if data successfully stored, false otherwise
+ */
+function saveMeeting() {
+  let allInputs = document.getElementsByTagName('input');
+  let radioGroups = new Set();
+  let radioChecked = new Set(); 
+  for (let i = 0; i < allInputs.length; i++) {
+    let input = allInputs.item(i);
+    let key = input.name;
+    let value = input.value;
+    // If this is a radio button, add to the set of radio 
+    // groups if not there already.
+    if (input.type === 'radio') {
+      if (!radioGroups.has(input.name)) {
+        radioGroups.add(input.name);
+      }
+      // If the radio button is checked, add to checked set
+      if (input.checked) {
+        radioChecked.add(input.name);
+        sessionStorage.setItem(key, value);
+      }
+    }
+    else { // all other types of inputs
+      if (value !== '') {
+        sessionStorage.setItem(key, value);
+      }
+      else {
+        alert(FILL_ALL_FIELDS_ALERT);
```

Comment:
Hmm, I'm not sure if it's the right thing to add UI calls here. It might be better to return a status code and produce the UI at a higher layer instead. In particular, this function might be fiddlier to test if it's producing UI inside your test harness.

Reply:
Is there a best practices on defining my own error codes? I'm thinking they will be integers (stored in a `const` variable), but what number do I start from 😅?

Reply:
Maybe an Error object is the right thing? http://go/mdn/JavaScript/Reference/Global_Objects/Error

Reply:
Though as status code as a string might be OK too since it's a lot more lightweight.

Reply:
Yeah, I was actually just looking into that and you messaged - I think I can maybe do something like `[throw](https://www.w3schools.com/js/js_errors.asp)(error message string)` and then catch and handle it at a higher level

Reply:
Hmm.. the spirit of the function is to validate some data, not to try an action that could fail. I think it's OK to do status codes. I would avoid throwing anything that's not an Error object because the js style guide suggests otherwise: go/js-style#features-exceptions

Reply:
I decided to go with `Error` objects, because if i returned a status code, I will need some way to map this status code to an error message in order to alert() the user, which would involve a lot of hard-coding that would probably be a pain to modify later on. Returning the entire error message string didn't seem like good practice either.

Currently, I just use `throw(new Error(error message))`, but I could also define my own error classes that are subclasses of `Error`. But the only thing I will modify in those subclasses will be their error message, so I don't know if it is worth it - what do you think?